### PR TITLE
Match collision functions

### PR DESCRIPTION
**Pull request by Unnunu**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 13 functions (+0.30%) on `us`.

✅ This PR fixes 1 warnings!

File: src/collision.c [hunk 1]
```diff
@@ -0,0 +1,1173 @@
...
+s32 collision_heap_create(void);
+void* collision_heap_malloc(s32 size);
+void collision_heap_free(void*);
+
+void load_hit_data(s32 idx, HitAsset* hit);
+void _add_hit_vert_to_buffer(Vec3f** buf, Vec3f* vert, s32* bufSize);
+s32 _get_hit_vert_index_from_buffer(Vec3f** buffer, Vec3f* vert, s32* bufferSize);
+
+void backup_map_collision_data(void)
+{
```
Comment (nanaian) [inline src/collision.c:64]: Hia, we use this brace style:
```suggestion
void backup_map_collision_data(void) {
```

File: src/collision.c [hunk 2]
```diff
@@ -0,0 +1,1173 @@
...
+    CollisionData *pCollisionData;
```
Comment (nanaian) [inline src/collision.c:65]: Align pointer asterisks to the left:
```suggestion
    CollisionData* pCollisionData;
```

File: src/collision.c [hunk 3]
```diff
@@ -0,0 +1,1173 @@
...
+        pVertices->y = pAssetVertices->y;
+        pVertices->z = pAssetVertices->z;
+    }
+
+    pAssetCollider = (HitAssetCollider*)((void*)hit + pAssetCollisionData->collidersOffset);
+    pCollider = pCollisionData->colliderList = collision_heap_malloc(pAssetCollisionData->numColliders * sizeof(Collider));
+    pCollisionData->numColliders = pAssetCollisionData->numColliders;
+    for (i = 0; i < pAssetCollisionData->numColliders; pAssetCollider++, pCollider++, i++)
+    {
+
```
Comment (nanaian) [inline src/collision.c:241]: Empty line here not needed

File: src/collision.c [hunk 4]
```diff
@@ -0,0 +1,1173 @@
...
+        pCollider->flags = 0;
+        pCollider->nextSibling = pAssetCollider->nextSibling;
+        pCollider->firstChild = pAssetCollider->firstChild;
+        pCollider->numTriangles = pAssetCollider->numTriangles;
+
+        numTrianles = pCollider->numTriangles;
```
Comment (nanaian) [inline src/collision.c:247]: Typo
```suggestion
        numTriangles = pCollider->numTriangles;
```

**Review by nanaian — CHANGES_REQUESTED** [review]
Awesome decomp & documentation work, especially for a first contribution! Thank you!
I've made a few comments on where your style deviates from the existing codebase - if you could fix them across `collision.c` that'd be great

File: src/collision.c [hunk 5]
```diff
@@ -219,15 +210,15 @@ void load_hit_data(s32 idx, HitAsset *hit) {
 
     pAssetBoundingBox = (u32*)((void*)hit + pAssetCollisionData->boundingBoxesOffset);;
     pCollisionData->aabbs = collision_heap_malloc(pAssetCollisionData->boundingBoxesDataSize * 4);
-    for (i = 0, pBoundingBox = (u32*)(pCollisionData->aabbs); i < pAssetCollisionData->boundingBoxesDataSize; pAssetBoundingBox++, pBoundingBox++, i++)
-    {
+    for (i = 0, pBoundingBox = (u32*)(pCollisionData->aabbs); i < pAssetCollisionData->boundingBoxesDataSize;
+         pAssetBoundingBox++, pBoundingBox++, i++) {
```
Comment (nanaian) [inline src/collision.c:214]: Looks like you have an extra space of indentation here

File: src/collision.c [hunk 6]
```diff
@@ -219,15 +210,15 @@ void load_hit_data(s32 idx, HitAsset *hit) {
...
         *pBoundingBox = *pAssetBoundingBox;
     }
 
     pAssetVertices = (Vec3s*)((void*)hit + pAssetCollisionData->verticesOffset);
     pCollisionData->vertices = collision_heap_malloc(pAssetCollisionData->numVertices * sizeof(Vec3f));
-    for (i = 0, pVertices = pCollisionData->vertices; i < pAssetCollisionData->numVertices; pVertices++, pAssetVertices++, i++)
-    {
+    for (i = 0, pVertices = pCollisionData->vertices; i < pAssetCollisionData->numVertices;
+         pVertices++, pAssetVertices++, i++) {
```
Comment (nanaian) [inline src/collision.c:221]: & here

**Review by nanaian — CHANGES_REQUESTED** [review]

**Review by nanaian — APPROVED** [review]

File: src/collision.c [hunk 7]
```diff
@@ -0,0 +1,1084 @@
+#include "common.h"
+#include "model.h"
+
+extern s16 D_800D91DC;
+
+typedef struct {
+    u32 collisionOffset;
+    u32 zoneOffset;
+} HitAsset;
```
Comment (ethteck) [inline src/collision.c:9]: Having offset / size comments for these would be nice, as we do for other structs (see common_structs.h)

File: src/collision.c [hunk 8]
```diff
@@ -0,0 +1,1084 @@
...
+
+        for (i = 0; i < collider->numTriangles; i++)
+            if (test_ray_triangle_horizontal(triangleTable++, collisionData->vertices))
+                ret = gCollisionRayLength;
+    }
+
+    return ret;
+}
+
+typedef struct {
```
Comment (ethteck) [inline src/collision.c:869]: This struct def and these externs should be at the top with the others (externs in ram order too please)

**Review by ethteck — CHANGES_REQUESTED** [review]
Amazing work, and welcome to the project! Just a few more style things I noticed. also, I'm asking @HailSanta to look over name changes and make sure he agrees with them or prefers different / the existing names he has.

By the way, what does the p prefix for locals that you're using mean?

**Comment by Unnunu** [comment]
Put struct defs and externs to the right place and also corrected the names. And p is for pointer. Should I remove it?

**Comment by ethteck** [comment]
Yeah, I'd remove it since we don't do that elsewhere and I find those kinds of prefixes more annoying than useful in practice, personally. Thanks for making the other changes!

**Pull request merged**